crates/driver/src/tests/setup/driver.rs

```diff
@@ -246,8 +248,26 @@ async fn create_config_file(
                 || "".to_string(),
                 |timestamp| format!("rank-by-surplus-date = \"{}\"", timestamp.to_rfc3339())
             ),
+            protocol_fee_discounted_addresses_to_toml(&solver.protocol_fee_discounted_addresses),
         )
         .unwrap();
     }
     file.into_temp_path()
 }
+
+fn protocol_fee_discounted_addresses_to_toml(
+    protocol_fee_discounted_addresses: &ProtocolFeeDiscountedAddresses,
+) -> String {
+    let mut result = String::new();
+    for (address, kind) in protocol_fee_discounted_addresses {
```

Comment:
Not a fan of reallocating the String `result` for each iteration, but since it is for test cases and we rarely use it, I think it is fine.